shared/arkitekt-decompose/src/commonMain/kotlin/app/futured/arkitekt/decompose/ext/DecomposeValueExt.kt

```diff
@@ -25,9 +28,16 @@ fun <T : Any> Value<T>.asFlow(): Flow<T> = callbackFlow {
 
 /**
  * Converts this Decompose [Value] to Kotlin [StateFlow].
+ *
+ * @param T The type of the value.
+ * @param coroutineScope The [CoroutineScope] in which the [StateFlow] is active.
+ * @return A [StateFlow] emitting the values of the [Value].
  */
-fun <T : Any> Value<T>.asStateFlow(coroutineScope: CoroutineScope): StateFlow<T> = asFlow().stateIn(
-    scope = coroutineScope,
-    started = SharingStarted.Lazily,
-    initialValue = value,
-)
+fun <T : Any> Value<T>.asStateFlow(
```

Comment:
JFY, this extension could be implemented without requiring `CoroutineScope`.

```kotlin
import com.arkivanov.decompose.value.Value
import kotlinx.coroutines.flow.FlowCollector
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

fun <T : Any> Value<T>.asStateFlow(): StateFlow<T> = ValueStateFlow(this)

private class ValueStateFlow<out T : Any>(
    private val valueRef: Value<T>,
) : StateFlow<T> {
    override val value: T get() = valueRef.value
    override val replayCache: List<T> get() = listOf(value)

    override suspend fun collect(collector: FlowCollector<T>): Nothing {
        val stateFlow = MutableStateFlow(value)
        val cancellation = valueRef.subscribe { stateFlow.value = it }

        try {
            stateFlow.collect(collector)
        } finally {
            cancellation.cancel()
        }
    }
}
```

Reply:
Oh, that's nice approach. Thanks for suggestion!